data-prepper-plugins/dynamodb-source-coordination-store/src/test/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbSourceCoordinationStoreTest.java

```diff
@@ -135,14 +135,26 @@ void tryUpdateSourcePartitionItem_calls_dynamoClientWrapper_correctly_for_differ
             final Instant partitionOwnershipTimeout = Instant.now();
             given(updateItem.getPartitionOwnershipTimeout()).willReturn(partitionOwnershipTimeout);
             doNothing().when((DynamoDbSourcePartitionItem)updateItem).setPartitionPriority(partitionOwnershipTimeout.toString());
+            createObjectUnderTest().tryUpdateSourcePartitionItem(updateItem);
         } else if (sourcePartitionStatus.equals(SourcePartitionStatus.CLOSED.toString())) {
             final Instant reOpenAtTime = Instant.now();
             given(updateItem.getReOpenAt()).willReturn(reOpenAtTime);
             doNothing().when((DynamoDbSourcePartitionItem) updateItem).setPartitionPriority(reOpenAtTime.toString());
+            createObjectUnderTest().tryUpdateSourcePartitionItem(updateItem);
+        } else if (sourcePartitionStatus.equals(SourcePartitionStatus.COMPLETED.toString())) {
```

Comment:
Generally, different assertions should use different tests. Normally, having conditionals in a test in an indication that the testing approach may not be ideal.

Reply:
I will separate these tests based on the status